File: extra_1/app/src/main/java/com/example/extra_1/api/WeatherApi.java
package com.example.extra_1.api;

import com.example.extra_1.model.WeatherModels;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface WeatherApi {
    @GET("data/2.5/forecast")
    Call<WeatherModels.ForecastResponse> getForecast(
            @Query("q") String city,
            @Query("appid") String apiKey,
            @Query("units") String units,
            @Query("lang") String lang
    );
}

File: extra1/app/src/main/java/com/example/extra1/api/WeatherApi.kt
package com.example.extra1.api

import com.example.extra1.model.ForecastResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface WeatherApi {
    @GET("data/2.5/forecast")
    fun getForecast(
        @Query("q") city: String,
        @Query("appid") apiKey: String,
        @Query("units") units: String = "metric",
        @Query("lang") lang: String = "ru"
    ): Call<ForecastResponse>
}
